### PR TITLE
front: use editoast /stdcm endpoint

### LINKED ISSUE #4743: Call the new endpoint STDCM in front end
# Description

The STDCM endpoint has been migrated in #4010 . We now need to use the new endpoint.

# Goal

Enables drop API

# Acceptance criteria

- [ ] The former STDCM endpoint in API isn't used anymore

### PR DESCRIPTION
**Pull request by anisometropie**
closes #4743

### CODE REVIEW

File: front/src/common/api/osrdEditoastApi.ts [hunk 1]
```diff
@@ -1171,7 +1171,25 @@ export type PostStdcmApiResponse =
...
+    maximum_departure_delay?: number;
+    maximum_run_time?: number;
+    rolling_stock_id: number;
+    rolling_stocks: number[];
+    speed_limit_tags?: string;
+    standard_allowance: AllowanceValue;
+    start_time?: number;
+    steps: {
+      duration?: number;
+      waypoints?: Waypoint;
```
Comment (clarani) [inline front/src/common/api/osrdEditoastApi.ts]: I think duration and waypoints are required
```suggestion
      duration: number;
      waypoints: Waypoint;
```

File: front/src/applications/stdcm/formatStcmConf.ts [hunk 2]
```diff
@@ -90,40 +88,39 @@ export default function formatStdcmConf(
   // Demander: format de date
   // Demander: pourquoi tableaux
```
Comment (clarani) [inline front/src/applications/stdcm/formatStcmConf.ts]: Could you remove these comments please ? :) 
```suggestion
```

File: front/src/applications/stdcm/formatStcmConf.ts [hunk 3]
```diff
@@ -1,28 +1,26 @@
...
   setFailure: (e: Error) => ThunkAction<ActionFailure>,
   t: TFunction,
   osrdconf: OsrdStdcmConfState
-) {
+): PostStdcmApiArg | undefined {
   const { isByOrigin, isByDestination } = makeEnumBooleans(STDCM_MODES, osrdconf.stdcmMode);
 
   let error = false;
-  if (!osrdconf.origin && isByOrigin) {
+  if (!(osrdconf.origin && osrdconf.originUpperBoundTime) && isByOrigin) {
```
Comment (clarani) [inline front/src/applications/stdcm/formatStcmConf.ts]: Why did you add the check on originUpperBoundTime here ? If it is undefined, I'm not sure that you will display the right message

**Review by clarani — COMMENTED** [review]
Almost good for me, only left 3 small comments :)

File: front/src/applications/stdcm/formatStcmConf.ts (hunk 3 above)
Comment (anisometropie) [reply front/src/applications/stdcm/formatStcmConf.ts]: that was supposed to be for `originTime`.
both `originTime` and `originUpperBoundTime` are used to get `start_time`

Screenshot_20230821_123118

Comment (anisometropie) [reply front/src/applications/stdcm/formatStcmConf.ts]: fixed

**Review by clarani — APPROVED** [review]
LGTM and tested in local :heavy_check_mark:

**Review by flomonster — APPROVED** [review]
LGTM. (reviewed editoast)
